### PR TITLE
[SPARK-11667] Update dynamic allocation docs to reflect supported cluster managers

### PR DESCRIPTION
**Pull request by andrewor14**

### CODE REVIEW

**Comment by andrewor14** [comment]
@tnachen please verify whether the mesos part is correct.

File: docs/job-scheduling.md [hunk 1]
```diff
@@ -56,36 +56,31 @@ provide another approach to share RDDs.
...
+In standalone mode, simply start your workers with `spark.shuffle.service.enabled` set to `true`.
 
-Additionally, your application must use an external shuffle service. The purpose of the service is
-to preserve the shuffle files written by executors so the executors can be safely removed (more
-detail described [below](job-scheduling.html#graceful-decommission-of-executors)). To enable
-this service, set `spark.shuffle.service.enabled` to `true`. In YARN, this external shuffle service
-is implemented in `org.apache.spark.yarn.network.YarnShuffleService` that runs in each `NodeManager`
-in your cluster. To start this service, follow these steps:
+In Mesos coarse-grained mode, run `$SPARK_HOME/sbin/start-mesos-shuffle-service.sh` on all
+slave nodes with `spark.shuffle.service.enabled` set to `true`.
```
Comment (tnachen) [inline docs/job-scheduling.md]: I'd like to add that users can run the mesos-shuffle-service.sh with Marathon, and they should start the service in the foreground running `spark-class org.apache.spark.deploy.mesos.MesosExternalShuffleService`

**Comment by tnachen** [comment]
Only have one comment, otherwise everything else LGTM

File: docs/job-scheduling.md (hunk 1 above)
Comment (andrewor14) [inline docs/job-scheduling.md]: not sure what you mean about the latter part. If you wish you can submit a patch against the same issue to augment the description here.

**Comment by tnachen** [comment]
I think this is fine for now, I was thinking leaving some comments about how it should be launched with Marathon but I think later we can add a example json in the Marathon repo that we can link from here.

**Comment by tnachen** [comment]
LGTM

**Comment by andrewor14** [comment]
Thanks Tim, merging into master 1.6.

**Pull request closed without merging**